src/backend/src/mavsdk_server.cpp

```diff
@@ -2,33 +2,50 @@
 #include <iostream>
 #include <string>
 
-static void usage();
-
 static auto constexpr default_connection = "udp://:14540";
+static auto default_mavsdk_server_port = 0;
+
+static void usage();
+static bool is_integer(const std::string& tested_integer);
 
 int main(int argc, char** argv)
 {
-    if (argc > 2) {
-        usage();
-        return 1;
-    }
+    std::string connection_url = default_connection;
+    int mavsdk_server_port = default_mavsdk_server_port;
+
+    for (int i = 1; i < argc; i++) {
+        const std::string current_arg = argv[i];
 
-    if (argc == 2) {
-        const std::string positional_arg = argv[1];
-        if (positional_arg.compare("-h") == 0 || positional_arg.compare("--help") == 0) {
+        if (current_arg == "-h" || current_arg == "--help") {
             usage();
             return 0;
+        } else if (current_arg == "-p") {
+            if (argc <= i + 1) {
+                usage();
+                return 1;
+            }
+
+            const std::string port(argv[i + 1]);
+            i++;
+
+            if (!is_integer(port)) {
+                usage();
+                return 1;
+            } else {
```

Comment:
I prefer no `else` but that's just a nit.

Reply:
Done